qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,121 @@
+    1). run query cmd. e.g -cpu ?cpuid
+    2). check the expected info is included in the cmd output.
+    3). Boot guest and check the output of qmp command "qom-list-types"
+    4). Check the output of qmp command "query-cpu-definitions"
+    5). Check the output of qmp command "query-cpu-model-expansion"
+    """
+
+    def remove_models(model_list):
+        """
+        Remove models from cpu_types
+        :param model_list The list of models to be removed
+        """
+        for model in model_list:
+            try:
+                cpu_types.remove(model)
+            except ValueError:
+                logging.warning('The model to be removed is not in the list: %s' % model)
+                continue
+
+    cpu_types = []
+    list(map(cpu_types.extend, list(cpu.CPU_TYPES.values())))
+
+    qemu_path = utils_misc.get_qemu_binary(params)
+    qemu_version = env_process._get_qemu_version(qemu_path)
+    match = re.search(r'[0-9]+\.[0-9]+\.[0-9]+(\-[0-9]+)?',
+                      qemu_version)
+    host_qemu = match.group(0)
+    if host_qemu in VersionInterval('[, 5.0.0)'):
+        remove_models(params.objects('cpu_model_5_0_0'))
+    if (host_qemu in VersionInterval('[,4.2.0)') and not
+            params.get('os_variant') == 'rhel8'):
+        remove_models(params.objects('cpu_model_4_2_0'))
+        if params.get('os_app_variant') == '0':
+            remove_models(params.objects('cpu_model_8_1_0'))
+    if host_qemu in VersionInterval('[,3.1.0)'):
+        remove_models(params.objects('cpu_model_3_1_0'))
+
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    logging.info('Query cpu models by qemu command')
+    query_cmd = "%s -cpu ? | awk '{print $2}'" % qemu_binary
+    qemu_binary_output = process.system_output(
+        query_cmd, shell=True).decode().splitlines()
+    cpuid_index = qemu_binary_output.index('CPUID')
+    cpu_models_binary = qemu_binary_output[1: cpuid_index - 1]
+    cpu_flags_binary = qemu_binary_output[cpuid_index + 1:]
+    params['start_vm'] = 'yes'
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    # query cpu model supported by qemu
+    logging.info('Query cpu model supported by qemu by qemu monitor')
+    qmp_model_output = str(vm.monitor.cmd('qom-list-types'))
+    qmp_def_output = str(vm.monitor.cmd('query-cpu-definitions'))
+
+    # Check if all the output contain expected cpu models
+    missing = [[], [], []]
+    output_list = [cpu_models_binary, qmp_model_output, qmp_def_output]
+    for cpu_model in cpu_types:
+        logging.info('Check cpu model %s from qemu command output and'
+                     ' qemu monitor output' % cpu_model)
+        for index, output in enumerate(output_list):
+            if cpu_model not in output:
+                missing[index].append(cpu_model)
+    for index, rec in enumerate(missing):
+        if rec:
+            test.fail('%s is missing in the output\n %s' %
```

Comment:
Prefer a clearer log to tell which output miss the flag.

Reply:
@huangyum Updated, please help re-review, thanks.